ansible/library/testbed_vm_info.py

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env python
+
+import re
+import yaml
+import os
+import traceback
+import subprocess
+import ipaddr as ipaddress
+from operator import itemgetter
+from itertools import groupby
+from collections import defaultdict
+
+DOCUMENTATION='''
+module: testbed_vm_info.py
+Ansible_version_added:  2.0.0.2
+short_description:   Gather all related VMs info 
+Description:
+       When deploy testbed topology with VM connected to SONiC, gather neighbor VMs info for generating SONiC minigraph file
+ options:
+    base_vm:  base vm name defined in testbed.csv for the deployed topology; required: True
+    topo:     topology name defined in testbed.csv for the deployed topology; required: True
+
+Ansible_facts:
+    'neighbor_eosvm_mgmt':  all VM hosts management IPs 
+    'topoall':              topology information
+
+'''
+
+EXAMPLES='''
+    - name: gather vm information
+      testbed_vm_info: base_vm='VM0100' topo='t1'
+'''
+
+### Here are the assumption/expectation of files to gather VM informations, if the file location or name changes, please modify it here 
+TOPO_PATH = 'vars/'
+VM_INV_FILE = 'veos'
+
+
+class TestbedVMFacts():
+    """
+    Retrieve testbed VMs management information that for a specified toplogy defined in testbed.csv
+
+    """
+
+    def __init__(self, toponame, vmbase):
+        self.topofile = TOPO_PATH+'topo_'+toponame +'.yml'
+        self.start_index = int(re.findall('VM(\d+)', vmbase)[0])
+        self.vmhosts = {}
+        return
+
+
+    def get_neighbor_eos(self):
+        eos = {}
+        with open(self.topofile) as f:
+            vm_topology = yaml.load(f)
+            f.close()
```

Comment:
'with' keyword will close the 'f' file descriptor for you. so it's better to remove f.close() from here